src/budget/NewCategoryPage.js

```diff
@@ -45,13 +64,38 @@ export default function NewCategoryPage() {
           <form>
             <div className="mb-3">
               <label htmlFor="formName" className="form-label">Name</label>
-              <input type="text" onChange={(event) => setName(event.target.value)} className="form-control" name="formName" id="formName" />
+              <input
+                type="text"
+                onChange={handleNameChange}
+                className={(errors.name.length !== 0) ? 'form-control border-danger' : 'form-control'}
+                name="formName"
+                id="formName"
+                required
+              />
+              {errors && <div className="text-danger">{errors.name.map((message) => <div className="pt-1" key={message}>{message}</div>)}</div>}
             </div>
             <div className="mb-3">
               <label htmlFor="formTarget" className="form-label">Target</label>
-              <input type="text" onChange={handleTargetChange} className="form-control" name="formTarget" id="formTarget" />
+              <input
+                type="text"
+                onChange={handleTargetChange}
+                className={(errors.target.length !== 0) ? 'form-control border-danger' : 'form-control'}
+                name="formTarget"
+                id="formTarget"
+                required
+              />
+              {errors && <div className="text-danger">{errors.target.map((message) => <div className="pt-1" key={message}>{message}</div>)}</div>}
             </div>
```

Comment:
The error lines are still quite long and a bit hard to read. Can we break each opening and closing tag onto their own lines?